### PR TITLE
Slimline version of github commit refactor

### LINKED ISSUE #764: Github commit count on profile page
imported from https://www.pivotaltracker.com/story/show/64891298

As a member of AV
So that I can find out what projects others are contributing to
I should see the number of commits they have made to projects listed on their profile page.

### PR DESCRIPTION
**Pull request by tansaku**
fixes #764 

@sampritipanda and I cherry picked various components from the original pull request, so this is now cleanly focused just on the github commit refactor

### CODE REVIEW

File: app/jobs/github_commits_job.rb [hunk 1]
```diff
@@ -1,63 +1,58 @@
...
-            puts 'Something is Wrooong!'.yellow
-          end
-        rescue Exception
-          puts "#{user.display_name} will not stop me!".red
+
+    contributors.map do |contributor|
+      begin
+        user = User.find_by_github_username(contributor.author.login)
+
+        if user
```
Comment (tansaku) [inline app/jobs/github_commits_job.rb]: could be shortened with some guard clause ...?

File: app/jobs/github_commits_job.rb [hunk 2]
```diff
@@ -1,63 +1,58 @@
...
-    begin
-      doc = Nokogiri::HTML(open(project.github_url))
-      commit_count = doc.at_css('li.commits .num').text.strip.gsub(',', '').to_i
-      project.update(commit_count: commit_count)
-      puts "Got what I came for from #{project.github_url}".green
-    rescue Exception
-      puts "I refuse to fail or be stopped by #{project.github_url}!".red
+    repo_commits = Array.new
+
+    client.contributors("#{project.github_repo_name}/#{project.github_repo_user_name}", true, per_page: 100).each do |contrib|
```
Comment (tansaku) [inline app/jobs/github_commits_job.rb]: could use reduce here ...

File: app/jobs/github_commits_job.rb [hunk 3]
```diff
@@ -1,63 +1,58 @@
...
+      repo_commits << contrib.contributions
     end
+
+    commit_count = repo_commits.reduce(:+)
+    project.update(commit_count: commit_count)
```
Comment (tansaku) [inline app/jobs/github_commits_job.rb]: could use reduce directly here ...

```ruby
  def update_total_commit_count_for(project)
    commit_count = client.contributors(github_url(project), true, per_page: 100).reduce(0) do |memo, contrib|
      memo += contrib.contributions
    end
    project.update(commit_count: commit_count)
  end

  def github_url(project)
    "#{project.github_repo_name}/#{project.github_repo_user_name}"
  end
```
and this github_url method I've extracted could be pulled into project I think ...

**Comment by tansaku** [comment]
I see the tests are wrapped with VCR, however as a unit test, we might prefer dependency injection of a mock, but that's not specific to this pull request itself and shouldn't be a blocker per se, since this pull request is definitely a step in the right direction ...

In LocalSupport we're trying to only use VCR/webmock in acceptance tests and dependency inject mocks into unit tests (as well as removing FactoryGirl for speed where possible) - although this job is a standlone entity that arguably deserves a full integration ... hmm ... @diraulo will be interested in your thoughts ...

**Comment by tansaku** [comment]
@diraulo I have an all green version of the suggested code changes I made ready to push up into this pull request.  Am just holding off so you can see the before and after ...

File: config/settings/production.yml [hunk 4]
```diff
@@ -34,3 +34,5 @@ twitter:
   access_token: <%= ENV["TWITTER_ACCESS_TOKEN"] %>
   access_token_secret: <%= ENV["TWITTER_TOKEN_SECRET"] %>
 
+github:
```
Comment (tansaku) [inline config/settings/production.yml]: should remove this

File: app/jobs/github_commits_job.rb (hunk 3 above)
Comment (diraulo) [inline app/jobs/github_commits_job.rb]: :+1:

**Comment by tansaku** [comment]
@diraulo I have a feeling this may have merge conflicts once @tochman's event update goes through - will be very interested in your thoughts on my refactorings - will fix up any merge conflicts tomorrow :-)

File: config/settings/production.yml (hunk 4 above)
Comment (diraulo) [inline config/settings/production.yml]: Yup! The global setting file should override this one.

File: features/users/github_commit_count.feature [hunk 5]
```diff
@@ -19,7 +19,7 @@ Feature: Displaying GitHub contribution statistics for user
   Scenario: Displays commit counts on user profile page
     Given I am on "profile" page for user "Bryan"
     Then I should see "Contributions"
-    And I should see "WebsiteOne - 388"
+    And I should see "WebsiteOne - 394"
```
Comment (diraulo) [inline features/users/github_commit_count.feature]: Just a something to keep in mind here. see my commit message https://github.com/diraulo/WebsiteOne/commit/f3565d04b023ca275894b10d8d8ac2da385c5be7 for more details

Comment (tansaku) [inline features/users/github_commit_count.feature]: @diraulo isn't the whole thing wrapped in VCR?  if not then it should be ...

Comment (diraulo) [inline features/users/github_commit_count.feature]: @tansaku It failed for me when I ran it locally. That's why I updated it to that value.

Comment (tansaku) [inline features/users/github_commit_count.feature]: @diraulo right, I guess the specs are wrapped in VCR, but not the features. I'll open a ticket for that --> #812

**Comment by tansaku** [comment]
@diraulo this should be good to go now - please merge in when you have a moment.

**Pull request merged**